File: KeeprFinalCheckpoint/Models/Keep.cs
namespace KeeprFinalCheckpoint.Models;

public class Keep
{
    public int Id { get; set; }
    public string CreatorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public int Views { get; set; }
    public int Kept { get; set; }
    public Profile Creator { get; set; }
}

// NOTE: this is the key to attaching the vaultKeepId to the keep when we do our dapper statement. Without this, the dapper statement is impossible to create for getting keeps in a vault.

public class VaultKeepViewModel : Keep
{
    public int VaultKeepId { get; set; }
    public string ProfileId { get; set; }
}

File: KeeprFinalCheckpoint.client/src/models/Keep.js


export class Keep{
  constructor(data){
    this.id = data.id
    this.creatorId = data.creatorId
    this.name = data.name
    this.description = data. description
    this.img = data.img
    this.views = data.views
    this.kept = data.kept
    this.creator = data.creator
  }
}

export class VaultKeep extends Keep{
  constructor(data){
    super(data)
    this.vaultKeepId = data.vaultKeepId
    this.vaultKeepCreatorId = data.profileId
  }
}